### PR TITLE
Refactor `Yamux::close` to use `?`

### PR DESCRIPTION
**Pull request by thomaseizinger**
# Description

(Hopefully) improves readability of the implementation due to reduced noise.

## Links to any relevant issues

Extracted out of https://github.com/libp2p/rust-libp2p/pull/2648.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: muxers/yamux/src/lib.rs [hunk 1]
```diff
@@ -179,16 +179,23 @@ where
...
+            .map_err(YamuxError)?
+        {
+            return Poll::Ready(Ok(()));
         }
-        while let std::task::Poll::Ready(x) = inner.incoming.poll_next_unpin(c) {
-            match x {
-                Some(Ok(_)) => {} // drop inbound stream
-                Some(Err(e)) => return Poll::Ready(Err(e)),
+
+        while let Poll::Ready(_inbound_stream) =
```
Comment (mxinden) [inline muxers/yamux/src/lib.rs]:
```suggestion
        while let Poll::Ready(inbound_stream) =
```
Why prefix `inbound_stream` with an `_`?

Comment (thomaseizinger) [reply muxers/yamux/src/lib.rs]: To make it more obvious that it will be unused but I guess _technically_ it is not unused!

Comment (thomaseizinger) [reply muxers/yamux/src/lib.rs]: I've tried to improve this with [6eb64fe](https://github.com/libp2p/rust-libp2p/pull/2677/commits/6eb64fe8155a6a18f7d67333fb9c43c3a50885cc).

**Review by mxinden — APPROVED** [review]
Thanks!

**Pull request merged**